Replace WEBKIT2_COMPILATION and JSC_COMPILATION with BUILDING_WEBKIT

mcatanzaro opened the pull request:
#### f4e1ac8eb7d26bd88254271451cf1eab89de3e16
<pre>
Replace WEBKIT2_COMPILATION and JSC_COMPILATION with BUILDING_WEBKIT
<a href="https://bugs.webkit.org/show_bug.cgi?id=244615">https://bugs.webkit.org/show_bug.cgi?id=244615</a>

Reviewed by Adrian Perez de Castro.

Trying to reduce more legacy WebKit2 branding, I first wanted to rename
WEBKIT2_COMPILATION to WEBKIT_COMPILATION. But that&apos;s almost redundant
with BUILDING_WEBKIT, which is used in more places. I don&apos;t think we
need both. We don&apos;t need a separate preprocessor definition to
distinguish between building Source/WebKit vs. other components of
WebKit because Source/WebKit does not have any export macros, and it
would be extremely confusing to try to maintain such a distinction
anyway given that there&apos;s no easy way to distinguish between &quot;WebKit&quot;
(the whole thing) vs. &quot;WebKit&quot; (Source/WebKit). Since we don&apos;t need it,
just simplify everything down to BUILDING_WEBKIT.

We can also define it in a global location so that it&apos;s no longer needed
in a bunch of platform-specific places. It&apos;s especially useful to have
defined under Tools/ when building tests, so that we can test private
functions, for instance. We don&apos;t want to restrict WebKit from accessing
its *own* internals!

* Source/JavaScriptCore/API/glib/JSCAutocleanups.h:
* Source/JavaScriptCore/API/glib/JSCClass.h:
* Source/JavaScriptCore/API/glib/JSCContext.h:
* Source/JavaScriptCore/API/glib/JSCDefines.h:
* Source/JavaScriptCore/API/glib/JSCException.h:
* Source/JavaScriptCore/API/glib/JSCOptions.h:
* Source/JavaScriptCore/API/glib/JSCValue.h:
* Source/JavaScriptCore/API/glib/JSCVersion.h.in:
* Source/JavaScriptCore/API/glib/JSCVirtualMachine.h:
* Source/JavaScriptCore/API/glib/JSCWeakValue.h:
* Source/JavaScriptCore/PlatformGTK.cmake:
* Source/JavaScriptCore/PlatformWPE.cmake:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/Scripts/glib/generate-api-header.py:
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.h:
* Source/WebKit/UIProcess/API/gtk/WebKitEnumTypes.h.template:
* Source/WebKit/UIProcess/API/gtk/WebKitPointerLockPermissionRequest.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h:
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.h:
* Source/WebKit/UIProcess/API/wpe/WebKitColor.h:
* Source/WebKit/UIProcess/API/wpe/WebKitEnumTypes.h.template:
* Source/WebKit/UIProcess/API/wpe/WebKitRectangle.h:
* Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewBackend.h:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypes.h.template:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMDefines.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMDocument.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMElement.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMObject.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.template:
* Source/cmake/OptionsCommon.cmake:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestUIClient.cpp:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/254039@main">https://commits.webkit.org/254039@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa4a15dc0a7afa572e314daac10b2f3af041b49a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87755 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31849 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18488 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96955 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/150801 "failed Failed to compile WebKit Reverted pull request changes (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91723 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/30209 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26293 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79880 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91698 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93374 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24405 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74500 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24403 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/91359 "Found 2 webkitpy python2 test failures: webkitpy.common.checkout.scm.scm_unittest.GitSVNTest.test_rebase_in_progress, webkitpy.common.checkout.scm.scm_unittest.SVNTest.test_svn_apply, Found 2 webkitpy python3 test failures: webkitpy.common.checkout.scm.scm_unittest.GitSVNTest.test_rebase_in_progress, webkitpy.common.checkout.scm.scm_unittest.SVNTest.test_svn_apply") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79363 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79523 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67250 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79558 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27895 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13344 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/73280 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27868 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14359 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26057 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2840 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29554 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37273 "Found 1 new test failure: fast/events/blur-remove-parent-crash.html") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/76110 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29452 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33635 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16875 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
r=me, the webkitpy EWS failure seems unrelated, FWIW.

mcatanzaro committed:
Replace WEBKIT2_COMPILATION and JSC_COMPILATION with BUILDING_WEBKIT
https://bugs.webkit.org/show_bug.cgi?id=244615

Reviewed by Adrian Perez de Castro.

Trying to reduce more legacy WebKit2 branding, I first wanted to rename
WEBKIT2_COMPILATION to WEBKIT_COMPILATION. But that's almost redundant
with BUILDING_WEBKIT, which is used in more places. I don't think we
need both. We don't need a separate preprocessor definition to
distinguish between building Source/WebKit vs. other components of
WebKit because Source/WebKit does not have any export macros, and it
would be extremely confusing to try to maintain such a distinction
anyway given that there's no easy way to distinguish between "WebKit"
(the whole thing) vs. "WebKit" (Source/WebKit). Since we don't need it,
just simplify everything down to BUILDING_WEBKIT.

We can also define it in a global location so that it's no longer needed
in a bunch of platform-specific places. It's especially useful to have
defined under Tools/ when building tests, so that we can test private
functions, for instance. We don't want to restrict WebKit from accessing
its *own* internals!

* Source/JavaScriptCore/API/glib/JSCAutocleanups.h:
* Source/JavaScriptCore/API/glib/JSCClass.h:
* Source/JavaScriptCore/API/glib/JSCContext.h:
* Source/JavaScriptCore/API/glib/JSCDefines.h:
* Source/JavaScriptCore/API/glib/JSCException.h:
* Source/JavaScriptCore/API/glib/JSCOptions.h:
* Source/JavaScriptCore/API/glib/JSCValue.h:
* Source/JavaScriptCore/API/glib/JSCVersion.h.in:
* Source/JavaScriptCore/API/glib/JSCVirtualMachine.h:
* Source/JavaScriptCore/API/glib/JSCWeakValue.h:
* Source/JavaScriptCore/PlatformGTK.cmake:
* Source/JavaScriptCore/PlatformWPE.cmake:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/Scripts/glib/generate-api-header.py:
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.h:
* Source/WebKit/UIProcess/API/gtk/WebKitEnumTypes.h.template:
* Source/WebKit/UIProcess/API/gtk/WebKitPointerLockPermissionRequest.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h:
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.h:
* Source/WebKit/UIProcess/API/wpe/WebKitColor.h:
* Source/WebKit/UIProcess/API/wpe/WebKitEnumTypes.h.template:
* Source/WebKit/UIProcess/API/wpe/WebKitRectangle.h:
* Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewBackend.h:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypes.h.template:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMDefines.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMDocument.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMElement.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMObject.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.template:
* Source/cmake/OptionsCommon.cmake:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestUIClient.cpp:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: https://commits.webkit.org/254039@main
